test_launch_ros/test/test_launch_ros/launch_frontend/test_node_frontend.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of how to parse an xml."""
+
+import io
+import pathlib
+import textwrap
+
+from launch import LaunchService
+from launch.launch_frontend import Parser
+
+from launch_ros.utilities import evaluate_parameters
+
+import pytest
+
+# Scaping the quote is needed in 'a_string' launch configuration, becuase of how the parser works.
+# TODO(ivanpauno): Check if it's possible to avoid that.
```

Comment:
@ivanpauno is all this quoting for `'a_string'` due to substitution grammar parsing? Is that bad for YAML markup?

Reply:
Yeah, the substitution grammar needs the quotes escaped. In YAML markup you have the same problem, but it's not particularly "bad" for it.
The yaml example is in the same file (below the xml example).